src/filters/composed_filter.class.ts

```diff
@@ -1,84 +1,83 @@
-//@ts-nocheck
-(function (global) {
-  'use strict';
-
-  var fabric = global.fabric || (global.fabric = {}),
-    filters = fabric.Image.filters,
-    createClass = fabric.util.createClass;
-
+// @ts-nocheck
+import * as filters from '.';
+import { TClassProperties } from '../typedefs';
+import { AbstractBaseFilter, BaseFilter } from './base_filter.class';
+import {
+  isWebGLPipelineState,
+  T2DPipelineState,
+  TWebGLPipelineState,
+} from './typedefs';
+/**
+ * A container class that knows how to apply a sequence of filters to an input image.
+ */
+export class Composed extends BaseFilter {
   /**
-   * A container class that knows how to apply a sequence of filters to an input image.
+   * A non sparse array of filters to apply
    */
-  filters.Composed = createClass(
-    filters.BaseFilter,
-    /** @lends fabric.Image.filters.Composed.prototype */ {
-      type: 'Composed',
+  subFilters: AbstractBaseFilter[];
 
-      /**
-       * A non sparse array of filters to apply
-       */
-      subFilters: [],
+  constructor({
+    subFilters = [],
+    ...options
+  }: Partial<
+    filters.BaseFilterOptions & { subFilters: AbstractBaseFilter[] }
+  > = {}) {
+    super(options);
+    this.subFilters = subFilters;
+  }
 
-      /**
-       * Constructor
-       * @param {Object} [options] Options object
-       */
-      initialize: function (options) {
-        this.callSuper('initialize', options);
-        // create a new array instead mutating the prototype with push
-        this.subFilters = this.subFilters.slice(0);
-      },
-
-      /**
-       * Apply this container's filters to the input image provided.
-       *
-       * @param {Object} options
-       * @param {Number} options.passes The number of filters remaining to be applied.
-       */
-      applyTo: function (options) {
-        options.passes += this.subFilters.length - 1;
-        this.subFilters.forEach(function (filter) {
-          filter.applyTo(options);
-        });
-      },
+  /**
+   * Apply this container's filters to the input image provided.
+   *
+   * @param {Object} options
+   * @param {Number} options.passes The number of filters remaining to be applied.
+   */
+  applyTo(options: TWebGLPipelineState | T2DPipelineState) {
+    if (isWebGLPipelineState(options)) {
```

Comment:
this is a change supposedly

Reply:
i think is fine. the code before was adding a useless passes property to the 2d pipeline.